### ISSUE TITLE
Use of requests call without `timeout` parameter

### ISSUE THREAD

**Issue by Sai-Suraj-27** [huggingface/diffusers#7730]
### Describe the bug

At few places in the codebase, where `requests.get()` is used, the `timeout` parameter is not specified.
https://github.com/huggingface/diffusers/blob/d1e3f489e909d351ea6abced05b0cdaf3a41be7a/src/diffusers/utils/loading_utils.py#L28

The `timeout` parameter is used to set the maximum time to wait for a response from the server. By omitting the `timeout` parameter, the program may hang indefinitely while awaiting a response.

From the [documentation](https://requests.readthedocs.io/en/latest/user/advanced/#timeouts), we can see that the value of timeout can be mentioned in seconds (like 5 seconds).

I think it's good to use the `timeout` parameter with a value of 5 or 10 seconds.

### Reproduction

https://github.com/huggingface/diffusers/blob/d1e3f489e909d351ea6abced05b0cdaf3a41be7a/src/diffusers/utils/loading_utils.py#L28
https://github.com/huggingface/diffusers/blob/d1e3f489e909d351ea6abced05b0cdaf3a41be7a/scripts/convert_vae_pt_to_diffusers.py#L124-L125

### System Info

This is a potential bug in the codebase, Irrespective of the python version, Diffusers version and platform.

### Who can help?

@sayakpaul

**Comment by sayakpaul** [comment]
@DN6 what do you think?

**Comment by DN6** [comment]
Yeah timeout is a good addition. @Sai-Suraj-27 Would you like to open a PR for this?

**Comment by Sai-Suraj-27** [comment]
> Yeah timeout is a good addition. @Sai-Suraj-27 Would you like to open a PR for this?

Thanks for looking into it 👍🏻. Yes, I will open PR fixing this.

**Comment by frankjoshua** [comment]
I was testing our error handling today. Yes a timeout would be great. v0.32.2

**Comment by asomoza** [comment]
the PR related to this issue got staled and never heard back from the author, if anyone from the community wants to take it and finish, we would appreciate a lot, just remember to also credit the original author of the PR.

**Comment by kghamilton89** [comment]
Hi @asomoza, I'd be happy to finish this up.

**Comment by asomoza** [comment]
thank you!, please go ahead.
